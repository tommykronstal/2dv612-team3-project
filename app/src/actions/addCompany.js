import { UPDATE_COMPANY_NAME } from './types';

export function updateCompanyName (companyName) {
  return {
      type: UPDATE_COMPANY_NAME,
      companyName: companyName
  };
}

