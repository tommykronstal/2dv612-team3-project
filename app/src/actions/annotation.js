import {UPDATE_ANNOTATION} from './types'

export const updateAnnotation = (annotationText, materialId) => ({
  type: UPDATE_ANNOTATION,
  annotationText,
  materialId
})
