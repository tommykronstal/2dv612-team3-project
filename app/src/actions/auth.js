import { SET_LOGGED_IN, LOGOUT } from './types'
import { getPayloadFromJwtToken } from '../lib/jwt'

export const setUserToLoggedIn = response => ({ 
  type: SET_LOGGED_IN, 
  token: response.token, 
  payload: getPayloadFromJwtToken(response.token) 
})

export const logout = () => ({ type: LOGOUT })
