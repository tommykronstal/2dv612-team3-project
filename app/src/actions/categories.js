import {FETCH_CATEGORIES, SET_CATEGORIES} from './types'

export const fetchCategories = (updateConfig) => ({
  type: FETCH_CATEGORIES,
  updateConfig
})

export const setCategories = categories => ({
  type: SET_CATEGORIES, categories
})
