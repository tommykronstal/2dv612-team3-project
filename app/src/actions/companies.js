import {ADD_COMPANY} from './types'

export const addCompany = (company) => ({
  type: ADD_COMPANY,
  payload: company
})
