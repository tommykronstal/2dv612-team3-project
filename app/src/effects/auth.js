import {takeEvery} from 'redux-saga/effects'
import {SET_LOGGED_IN, LOGOUT} from '../actions/types'
import {storeJwtToken, resetJwtToken} from '../lib/jwt'

export function *watchAuthenticationActions () {
  yield takeEvery(SET_LOGGED_IN, storeJWT)
  yield takeEvery(LOGOUT, resetJWT)
}

export function *storeJWT ({ token }) {
  yield storeJwtToken(token)
}

export function *resetJWT () {
  yield resetJwtToken()
}
