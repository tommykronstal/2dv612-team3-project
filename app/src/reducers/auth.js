import {SET_LOGGED_IN, LOGOUT} from '../actions/types'
import { getJwtToken, getPayloadFromJwtToken } from '../lib/jwt'

const token = getJwtToken()
const data = getPayloadFromJwtToken(token)

const defaultState = {
  isAuthenticated: !!token,
  jwt: token,
  email: data.email,
  firstName: data.firstName,
  role: data.role,
  userId: data.userId,
  companyId: data.companyId
}

export default (state = defaultState, action) => {
  switch (action.type) {
  case SET_LOGGED_IN:
    return {
      ...state,
      isAuthenticated: true,
      jwt: action.token,
      email: action.payload.email,
      firstName: action.payload.firstName,
      role: action.payload.role,
      userId: action.payload.userId,
      companyId: action.payload.companyId
    }
  case LOGOUT:
    return {
      isAuthenticated: false,
      jwt: null,
      email: null,
      firstName: null,
      role: null,
      userId: null,
      companyId: null
    }
  default:
    return state
  }
}
