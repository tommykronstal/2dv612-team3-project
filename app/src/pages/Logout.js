import React, {Component} from 'react'
import {logout} from '../actions/auth'
import {connect} from 'react-redux'
import {Redirect} from 'react-router-dom'
import CenteredForm from '../components/common/CenteredForm'
import Loading from '../components/common/Loading'

class Logout extends Component {

  componentWillMount() {
    this.props.logout()
  }

  render() {
    if (!this.props.auth.isAuthenticated) return <Redirect to='/login' />

    return (
      <CenteredForm>
        <Loading />
      </CenteredForm>
    )
  }
}

function mapDispatchToProps(dispatch) {
  return {
    logout: _ => dispatch(logout()),
  }
}

export default connect(({auth}) => ({auth}), mapDispatchToProps)(Logout)
