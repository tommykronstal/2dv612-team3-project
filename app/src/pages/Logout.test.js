import React from 'react'
import renderer from 'react-test-renderer'

import { Logout, mapStateToProps, mapDispatchToProps } from './Logout'
import { logout } from '../actions/auth'

describe('Logout', () => {
  it('renders correctly', () => {
    const auth = { isAuthenticated: true }
    const logout = jest.fn()

    const tree = renderer
      .create(<Logout auth={auth} logout={logout} />)
      .toJSON()

    expect(logout).toHaveBeenCalled()
      
    expect(tree).toMatchSnapshot()
  })

  describe('state', () => {
    it('should have access to auth state', () => {
      const state = {
        auth: { token: 'wooo' }
      }
    
      expect(mapStateToProps(state)).toEqual({ auth: state.auth })
    })
  })

  describe('actions', () => {
    it('should be able to dispatch logout actions', () => {
      const dispatch = jest.fn()
    
      const props = mapDispatchToProps(dispatch)
      props.logout()
    
      expect(dispatch).toBeCalledWith(logout())
    })
  })
})
