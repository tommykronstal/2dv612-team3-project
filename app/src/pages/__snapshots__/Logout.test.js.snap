// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders correctly 1`] = `
<div
  className="sc-bwzfXH fBchqp"
>
  <form
    className="sc-htpNat bcPhuh"
    onSubmit={[Function]}
  >
    <section
      className="sc-bdVaJa femPbx"
    >
      <div>
        <div
          className="sc-bxivhb jCzWjx"
          size={1}
        />
        <div
          className="sc-bxivhb iyclwS"
          size={1}
        />
        <div
          className="sc-bxivhb eXMUyY"
          size={1}
        />
      </div>
    </section>
  </form>
</div>
`;
